week4-ts/src/components/Header.tsx

```diff
@@ -0,0 +1,178 @@
+  const [isLocation, setIsLocation] = useState<boolean>(false);
+
+  const storeSearchHttpHandler = async (params: Params) => {
+    const { data } = await searchStore(params);
+
+    handleIsSearch(false);
+    handleResults(data.documents);
+  };
+
+  const handleMyLocation = () => {
+    if (!isLocation) {
+      new Promise((resolve) => {
+        navigator.geolocation.getCurrentPosition((currentPosition) => {
+          position.current = currentPosition.coords;
+          const params = {
+            y: position.current.latitude,
+            x: position.current.longitude,
+            radius: 1000,
+            query: '베이커리',
+          };
+          storeSearchHttpHandler(params);
+        });
+      });
+    } else {
+      const params = {
+        y: position.current.latitude,
+        x: position.current.longitude,
+        radius: 1000,
+        query: '베이커리',
+      };
+      storeSearchHttpHandler(params);
+    }
+  };
+
+  const handleInputDisabled = () => {
+    if (null !== searchRef.current) {
+      searchRef.current.disabled = !searchRef.current.disabled;
+      setIsLocation((prev) => !prev);
+    }
+  };
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setInput(e.target.value);
+  };
+  const handleSearchButton = (e: React.MouseEvent<HTMLButtonElement>) => {
+    handleIsSearch(true);
+    e.preventDefault();
+    if (null !== searchRef.current) {
+      if (!searchRef.current.disabled) {
+        const params: Params = {
+          query: input + ' ' + '베이커리',
+        };
+        storeSearchHttpHandler(params);
+      } else {
+        handleMyLocation();
+      }
+    }
+  };
+  return (
+    <StHeaderContainer>
+      <h1>🍰 빵수니가 져아 🍰 </h1>
+      <StHeaderWrapper>
+        <StSearchButton isChoice={isLocation} onClick={handleInputDisabled}>
+          현위치
+        </StSearchButton>
+        <StSearchLabel>우리 동네 </StSearchLabel>
+        <StSearchInput
+          ref={searchRef}
+          type="text"
+          onChange={handleInputChange}
+          value={input}
+          placeholder="지역을 입력해주세요"
+        />
+        <StSearchButton isChoice={isLocation} type="submit" onClick={handleSearchButton}>
+          검색
+        </StSearchButton>
+      </StHeaderWrapper>
+    </StHeaderContainer>
+  );
+}
+export default Header;
+
+interface StHeaderProps {
+  isChoice: boolean;
+}
+const StHeaderContainer = styled.header`
+  ${({ theme }) => {
+    return css`
+      display: flex;
+      flex-direction: column;
+      justify-content: center;
+      width: 100%;
+      height: 100%;
+      padding: 3rem;
+      background-color: ${theme.colors.header};
+
+      color: ${theme.colors.brown};
+    `;
+  }}
+`;
+const StHeaderWrapper = styled.div`
+  ${({ theme }) => {
+    return css`
+      display: flex;
+      justify-content: center;
+      align-items: center;
+      margin: 10px;
+      color: ${theme.colors.brown};
+      font-size: ${theme.fontSizes.form};
+    `;
+  }}
```

Comment:
오호 theme 을 이렇게 썼구나?? 
나는 아래처럼 쓰는데 어떤 것 같아ㅏ??

```suggestion
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 10px;
    color: ${({ theme }) => theme.colors.brown};
    font-size: ${({ theme }) => theme.fontSizes.form};
```

Reply:
밑에가 더 조은것가타,,! 쓰고싶은곳만 쓰면데자나!!! 난 두개다 해보면서 느껴써!